packages/pages/src/components/analytics/Analytics.ts

```diff
@@ -0,0 +1,192 @@
+import { TemplateProps } from "../../common/src/template/types";
+import { getRuntime, isProduction } from "../../util";
+import { AnalyticsMethods } from "./interfaces";
+import {
+  ConversionDetails,
+  CookieManager,
+  EntityPage,
+  PagesAnalyticsService,
+  providePagesAnalytics,
+  StaticPage,
+  Visitor,
+} from "@yext/analytics";
+
+/**
+ * The Analytics class creates a stateful facade in front of the \@yext/analytics
+ * Library's pagesAnalyticsProvider class. It takes in some data from the
+ * template configuration and uses it to provide configuration to the
+ * pagesAnalyticsProvider.
+ *
+ * Additionally, it provides handlers for controlling user opt-in for compliance
+ * requirements as well as for debugging, enabling conversion tracking, saving
+ * user identity information, and creating named analytics scopes for
+ * easy tagging.
+ *
+ * @public
+ */
+export class Analytics implements AnalyticsMethods {
+  private _optedIn: boolean;
+  private _conversionTrackingEnabled = false;
+  private _scopes: string[] = [];
+  private _cookieManager: CookieManager | undefined;
+  // TODO: create event queue for things to fire when optIn is called if we are not optedIn
+  private _analyticsReporter: PagesAnalyticsService | undefined;
+  private _pageViewFired = false;
+  private _enableDebugging = false;
+  constructor(
+    private templateData: TemplateProps,
+    requireOptIn?: boolean | undefined
+  ) {
+    this._optedIn = !requireOptIn;
+    this.makeReporter();
+    this.pageView(); // TODO: should I await this?
```

Comment:
You can't. Not sure you'll be able to do this.

https://stackoverflow.com/questions/43431550/async-await-class-constructor

Reply:
Hmm I do love the builder pattern for constructing JS classes, I can look at that in the future. It could also significantly clean up some of the configuration mess with inspecting the stream document.

Reply:
technically we don't actually need to await it. i was just wondering if it would be better to